python/mxnet/gluon/parameter.py

```diff
@@ -580,7 +580,12 @@ def get(self, name, **kwargs):
         else:
             for k, v in kwargs.items():
                 if hasattr(param, k) and getattr(param, k) is not None:
-                    assert v is None or v == getattr(param, k), \
+                    existing = getattr(param, k)
+                    if k == 'shape' and len(v) == len(existing):
+                        if all(d1 == 0 or d1 == d2 for d1, d2 in zip(v, existing)):
+                            continue
+
```

Comment:
This is to tolerate weight sharing when one side is not yet initialized.

Reply:
but we do need the weight sharing, right?

Reply:
yes, this is a fix to allow weight sharing.